tests/integration/targets/nxos_facts/tests/common/all_facts.yaml

```diff
@@ -21,4 +21,17 @@
       - result.ansible_facts.ansible_net_memfree_mb > 1
       - result.ansible_facts.ansible_net_memtotal_mb > 1
 
+- name: Collect list of available network resources for nxos
+  register: result
+  cisco.nxos.nxos_facts:
+    available_network_resources: true
+    gather_network_resources: all
+
+- name: Assert that correct available_network_resources returned
+  assert:
+    that:
+      - result.changed == false
+      - "{{ result['ansible_facts']['ansible_net_gather_network_resources'] | symmetric_difference(result['ansible_facts']['ansible_net_gather_network_resources']) |length\
```

Comment:
Are we comparing the same objects here?

Reply:
yeah, that was a mistake, updated. Thanks